### PR TITLE
Create AKS maintenance page

### PR DESCRIPTION
**Pull request by johnake**
### Context

We need to create an AKS maintenance page that can be replicated for other services - so that we can redirect and inform the users when their page is down.

### Changes proposed in this pull request

- Add maintenance pages

- Update Makefile

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Comment by johnake** [comment]
[Findpub: AKS maintenance page](https://trello.com/c/VDSM9esc/870-findpub-aks-maintenance-page)

File: maintenance_page/manifests/production/ingress_external_to_maintenance2.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: publish-teacher-training-courses.service.gov.uk
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: publish-teacher-training-courses.service.gov.uk
```
Comment (RMcVelia) [inline maintenance_page/manifests/production/ingress_external_to_maintenance2.yml:8]: www.find, here and line 4

File: maintenance_page/scripts/failover.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+kubectl apply -n ${NAMESPACE} -f maintenance_page/manifests/maintenance/service_maintenance.yml
+
+echo Create maintenance ingress
+kubectl apply -n ${NAMESPACE} -f maintenance_page/manifests/${CONFIG}/ingress_maintenance.yml
+
+### Change ingress ###
+# Remove if the service does not declare the external ingress i.e. null_host_header = false
+# https://github.com/DFE-Digital/terraform-modules/blob/101f2fa721661cbf0ad05e8105c3628900351008/domains/environment_domains/variables.tf#L15
+echo Configure external ingress to point at the maintenance app
+kubectl apply -n ${NAMESPACE} -f maintenance_page/manifests/${CONFIG}/ingress_external_to_maintenance.yml
```
Comment (RMcVelia) [inline maintenance_page/scripts/failover.sh:25]: think we also need to add another apply for ingress_external_to_maintenance2.yml here too

File: maintenance_page/manifests/production/ingress_external_to_main3.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: www.api.publish-teacher-training-courses.service.gov.uk
```
Comment (RMcVelia) [inline maintenance_page/manifests/production/ingress_external_to_main3.yml:4]: without the www, just api.publish

File: maintenance_page/manifests/production/ingress_external_to_main3.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: www.api.publish-teacher-training-courses.service.gov.uk
```
Comment (RMcVelia) [inline maintenance_page/manifests/production/ingress_external_to_main3.yml:8]: without the www, just api.publish

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**